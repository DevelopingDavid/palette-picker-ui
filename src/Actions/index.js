export const addPalette = (palette) => ({
  type: 'ADD_PALETTE',
  palette
})

export const toggleLocked = (color) => ({
  type: 'TOGGLE_LOCKED',
  color
})

export const makeNewProject = (project) => ({
  type: "MAKE_NEW_PROJECT",
  project
})

export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasError = (message) => ({
  type: "HAS_ERROR",
  message
})