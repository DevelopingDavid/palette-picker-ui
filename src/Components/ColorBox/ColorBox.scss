.color-container {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 120px;
}

.card {
  margin: 5px;
  .color-code {
    text-align: center;
  }
}

.lock-open,
.lock-closed {
  display: flex;
  justify-content: center;
  margin: 5px 0;
}