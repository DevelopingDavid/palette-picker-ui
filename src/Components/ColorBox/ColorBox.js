import React, { Component } from 'react';
import { connect } from 'react-redux'
import { toggleLocked } from '../../Actions'
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const styles = {
  card: {
    minWidth: 275,
    maxWidth: 290
  }
};

export class ColorBox extends Component {
  constructor() {
    super() 
    this.state = {
      locked: false
    }
  }

  toggleLock = () => {
    const hex = this.props.colorObject.hex
    this.props.toggleLockInStore(hex)
  }

  
  render() {
    const foundColor = this.props.currentPalette.find(colorObj => {
      return colorObj.hex === this.props.colorObject.hex
    })
    const { hex } = this.props.colorObject
    const { classes } = this.props;
    return (
      <Card className={`${classes.card} card`} 
        onClick={this.toggleLock}  
        >
        <CardContent style={{backgroundColor: hex, height: '12vw'}} className='color-container'>
        </CardContent>
        {
          foundColor.locked ? <i className="material-icons lock-closed">lock</i> : <i className="material-icons lock-open">lock_open</i>
        } 
        <Typography className='color-code' variant="h6" color="inherit" noWrap>
          { hex }
        </Typography>
      </Card>
    )
  }
}

ColorBox.propTypes = {
  classes: PropTypes.object.isRequired,
};

export const mapDispatchToProps = (dispatch) => ({
  toggleLockInStore: (hex) => dispatch(toggleLocked(hex))
});

export const mapStateToProps = (state) => ({
  currentPalette: state.currentPalette
})

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(ColorBox));