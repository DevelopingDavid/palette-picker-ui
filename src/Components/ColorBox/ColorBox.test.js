import React from 'react'
import { ColorBox, mapStateToProps, mapDispatchToProps } from './ColorBox';
import { createShallow } from '@material-ui/core/test-utils';
import { toggleLocked } from '../../Actions';

jest.mock('../../Actions');

let mockPalette = [
  { hex: "#FB5985", locked: false },
  { hex: "#FFFFFF", locked: false },
  { hex: "#FFFFFF", locked: false },
  { hex: "#FFFFFF", locked: false },
  { hex: "#FFFFFF", locked: false }
];

let mockColor = {
  hex:'#FB5985'
};

let mockClasses = {
  card: ''
};

let mockFunction = jest.fn();

describe('ColorBox', () => {
  let wrapper;
  let shallow;
  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(
      <ColorBox
        currentPalette={mockPalette}
        colorObject={mockColor}
        toggleLockInStore={mockFunction}
        classes={mockClasses} />
    );
  });

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call upon toggleLockInStore', () => {
    wrapper.instance().toggleLock();
    expect(mockFunction).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    it('should mapStateToProps', () => {
      const mockState = {}
      const upDatedState = {
        currentPalette: []
      }
      const firstState = mapStateToProps(mockState)
      expect(firstState).toEqual({})
      const newState = mapStateToProps(upDatedState)
      expect(newState).toEqual(upDatedState)
    })

    it('should dispatch toggleLockInStore', () => {
      const mockDispatch = jest.fn()
      const mockHex = '#fff'
      const actionToDispatch = toggleLocked(mockHex)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleLockInStore(mockHex)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})