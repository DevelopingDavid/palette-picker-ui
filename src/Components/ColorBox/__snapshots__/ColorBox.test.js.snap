// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox should match Snapshot 1`] = `ShallowWrapper {}`;
